### PR TITLE
Open Weather Map API data 

### PR DESCRIPTION
**Pull request by ShubhamShaswat**
This the initial commit for the assignments.The data is collected using open weather map API which contains the weather forecast for next 30 days for the cities around the world. The data collected are wind speed, humidity, temperature and pressure.
The cities are specified in a separated text file which can be modified depending on one's need.

Work in progress:

- [x] collecting geographical data such latitude and longitude in Geocoding steps, the current library in use is geocoder.
- [ ] client.py still incomplete
- [x] creating requirements.txt

Looking forward for the feedback.

### CODE REVIEW

**Comment by KayvanShah1** [comment]
We will soon start reviewing your scripts

File: open-weather-map-api/src/dependencies/cleaning/cleaning.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		
+	def check_missing_values(self,df):
+		"""check for missing values replace it with NaN"""
+		df.replace(' ', math.nan)
+		return df
+
+	def country_iso_code(self,df):
+		"""convert country codes to ISO 3166-1 alpha3"""
+		
+		df['country'] = df['country'].apply(lambda x : pycountry.countries.get(alpha_2=x).alpha_3 if (len(x) == 2) else pycountry.countries.get(name=x).alpha_3)
```
Comment (KayvanShah1) [inline open-weather-map-api/src/dependencies/cleaning/cleaning.py]: Please use code-formatter extensions and libraries available in VSCODE to format such long lines

Comment (ShubhamShaswat) [reply open-weather-map-api/src/dependencies/cleaning/cleaning.py]: thanks !! I will do it.

**Comment by gnommer** [comment]
code wise everything looks fine. except we are to generate and store the data and metadata as separate files. and the time series data is merged with the metadata. if they could be processed separately then it would be useful. as time series and metadata are stored in different storage and aug_ids would be used to fetch them

**Comment by ShubhamShaswat** [comment]
So the objective is to create a time series data as a separate file,and in my case that would be pressure, humidity and temp with their own unique `id` .This unique id would be same as `aud_id` of the metadata and further we will use it to fetch the required values from both of these files.Please correct me if I am wrong.
Meanwhile, I will modify the code to generate the required data.

**Comment by ShubhamShaswat** [comment]
all the data generated in each step can be found under directory [data](https://github.com/ShubhamShaswat/ts-mesh-pipeline/tree/shubham_shaswat-open_weather_map_data/data).

**Comment by KayvanShah1** [comment]
Are you available for a technical discussion today evening around 7-8 pm?

**Comment by ShubhamShaswat** [comment]
Yes, I am available around 8 PM. We can do the technical discussion today.

**Comment by ShubhamShaswat** [comment]
Yes, I am available around 8 PM. We can do the technical discussion today.

**Comment by KayvanShah1** [comment]
Hi Shubham,
Can you share your updated resume? We require that for the onboarding process and your offer letter.
You can email your updated resume to shahkayvan28@gmail.com and CC to cgeetacharan@gmail.com, gracealtree@berkeley.edu.
Ignore if already done.

**Comment by KayvanShah1** [comment]
The assignment has ended now so closing the PR.

**Pull request closed without merging**